pandas/tests/series/indexing/test_setitem.py

```diff
@@ -220,15 +220,9 @@ def test_setitem_slice(self):
     def test_setitem_slice_integers(self):
         ser = Series(np.random.randn(8), index=[2, 4, 6, 8, 10, 12, 14, 16])
 
-        msg = r"In a future version, this will be treated as \*label-based\* indexing"
-        with tm.assert_produces_warning(FutureWarning, match=msg):
-            ser[:4] = 0
-        with tm.assert_produces_warning(
-            FutureWarning, match=msg, check_stacklevel=False
-        ):
-            assert (ser[:4] == 0).all()
-        with tm.assert_produces_warning(FutureWarning, match=msg):
-            assert not (ser[4:] == 0).any()
+        ser[:4] = 0
```

Comment:
I might misunderstand this, but as I see it this would pass if both (__getitem__ and __setitem__) are label based? If yes, can you set explicitly to 0 during creation?

Reply:
good idea, updated